diff --git a/lib/Settings/SystemConfig.php b/lib/Settings/SystemConfig.php
index 7814f071..80fc7869 100644
--- a/lib/Settings/SystemConfig.php
+++ b/lib/Settings/SystemConfig.php
@@ -124,6 +124,13 @@ class SystemConfig
      */
     public static function get(string $key, mixed $default = null): mixed
     {
+        switch ($key) {
+          case "memories.exiftool": return "@exiftool@";
+          case "memories.vod.ffmpeg": return "@ffmpeg@";
+          case "memories.vod.ffprobe": return "@ffprobe@";
+          case "memories.vod.path": return "@go-vod@";
+        }
+
         if (!\array_key_exists($key, self::DEFAULTS)) {
             throw new \InvalidArgumentException("Invalid system config key: {$key}");
         }
@@ -154,6 +161,21 @@ public static function get(string $key, mixed $default = null): mixed
      */
     public static function set(string $key, mixed $value): void
     {
+        switch ($key) {
+          case "memories.exiftool":
+            $config->setSystemValue($key, @exiftool@);
+            return;
+          case "memories.vod.ffmpeg":
+            $config->setSystemValue($key, @ffmpeg@);
+            return;
+          case "memories.vod.ffprobe":
+            $config->setSystemValue($key, @ffprobe@);
+            return;
+          case "memories.vod.path":
+            $config->setSystemValue($key, @go-vod@);
+            return;
+        }
+
         // Check if the key is valid
         if (!\array_key_exists($key, self::DEFAULTS)) {
             throw new \InvalidArgumentException("Invalid system config key: {$key}");
